File: 프로그래머스/0/120869. 외계어 사전/외계어 사전.cs
using System;

public class Solution 
{
    public int solution(string[] spell, string[] dic) 
    {
        int count = 0;
        bool isContains = false;
        for(int i=0; i<dic.Length; i++)
        {
            for(int j=0; j<spell.Length; j++)
            {
                if(dic[i].Contains(spell[j]))
                {
                    count++;
                }
            }
            if(count == spell.Length) {
                isContains = true;
                break;
            }
            count = 0;
        }
        return isContains ? 1 : 2;
    }
}

File: 프로그래머스/0/120869. 외계어 사전/외계어 사전.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(vector<string> spell, vector<string> dic)
{
    int answer = 2;
    vector<string> vec;
    for(string &str : dic)
    {
        bool isBool = false;
        for(string &s : spell)
        {
            if(str.find(s) == string::npos)
            {   
                isBool = true;
            }
        }
        if(isBool)
        {
            isBool = false;
            continue;
        }
        vec.push_back(str);
    }
    
    for(string &str : vec)
    {
        for(string &s : spell)
        {
            while(str.find(s) != string::npos)
            {   
                size_t pos = str.find(s);
                str.erase(pos,1);
            }
        }
        cout << str << " ";
        if(str == "")
        {
            answer = 1;
            break;
        }
    }
    return answer;
}